src/commands/init.ts

```diff
@@ -134,95 +139,57 @@ export abstract class InitCommand extends BaseCommand {
     return parsed;
   }
 
+  private isNameValid(name: string) {
+    if (!name) {
+      return false;
+    }
+    return ComponentSlugUtils.Validator.test(name);
+  }
+
   async run(): Promise<void> {
     const { flags, args } = await this.parse(InitCommand);
 
-    const compose_exist = this.doesDockerComposeYmlExist();
-    if (!args.name) {
-      const answers = await inquirer.prompt([
-        {
-          type: 'input',
-          name: 'name',
-          message: 'What is the name of your project?',
-          when: !flags.name,
-          filter: value => value.toLowerCase(),
-          validate: (value) => {
-            if ((new RegExp('^[a-z][a-z-]+[a-z]$').test(value))) {
-              return true;
-            }
-            return `Component name can only contain lowercase letters and dashes, and must start and end with a letter.`;
-          },
-        },
-      ]);
-      args.name = answers.name;
-    }
+    const default_compose_file = this.getDefaultDockerComposeFile();
+    const invalid_name_message = 'Component name can only contain lowercase letters and dashes, and must start and end with a letter.';
 
-    while (!ComponentSlugUtils.Validator.test(args.name)) {
-      const init_type = (compose_exist || flags['from-compose']) ? 'Component' : 'Project';
-      const err_msg = `${init_type} name can only contain lowercase letters and dashes, and must start and end with a letter.`;
-      this.log(chalk.yellow(err_msg));
+    if (args.name && !this.isNameValid(args.name)) {
+      this.log(chalk.yellow(`Your project name is invalid. ${invalid_name_message} Please enter a new name.`));
+    }
 
-      const answers: any = await inquirer.prompt([
-        {
-          type: 'input',
-          name: 'name',
-          message: `Please provide a new name for your ${init_type.toLowerCase()}:`,
+    const answers = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'name',
+        message: 'What is the name of your project?',
+        when: !this.isNameValid(args.name),
+        filter: value => value.toLowerCase(),
+        validate: (value) => {
+          if (this.isNameValid(value)) {
+            return true;
+          }
+          return invalid_name_message;
         },
-      ]);
-      args.name = answers.name;
-    }
+      },
+    ]);
+    args.name = args.name || answers.name;
```

Comment:
Looks like `args.name` and `answers.name` should be flipped. This is the current UX when I start with a bad name

```
ryan@ryan-ThinkPad-P53s:~/Code/test-init$ architect init ----miltons-stapler----
Your project name is invalid. Component name can only contain lowercase letters and dashes, and must start and end with a letter. Please enter a new name.
? What is the name of your project? miltons-stapler
? We detected a docker-compose.yml file. Would you like to use it for your project? No
? Please select a framework/language for your project Nextjs

######################################
##### Let's set up your project! #####
######################################

Creating project directory... ✓
Command failed with exit code 129: git clone https://github.com/architect-templates/next ----miltons-stapler----
Error: Command failed with exit code 129: git clone https://github.com/architect-templates/next ----miltons-stapler----
    at makeError (/home/ryan/Code/architect-cli/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/home/ryan/Code/architect-cli/node_modules/execa/index.js:118:26)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Function.downloadGitHubRepos (/home/ryan/Code/architect-cli/src/architect/project/project.utils.ts:64:7)
    at async InitCommand.runProjectCreation (/home/ryan/Code/architect-cli/src/commands/init.ts:101:5)
    at async InitCommand.run (/home/ryan/Code/architect-cli/src/commands/init.ts:193:5)
    at async InitCommand._run (/home/ryan/Code/architect-cli/node_modules/@oclif/core/lib/command.js:80:22)
    at async Config.runCommand (/home/ryan/Code/architect-cli/node_modules/@oclif/core/lib/config/config.js:272:25)
    at async Object.run (/home/ryan/Code/architect-cli/node_modules/@oclif/core/lib/main.js:76:5)
```

Notice that the bad name gets carried through to the end of the prompts, rather than the actual name used in the prompt

Reply:
Great catch!